JenkinsFile

```diff
@@ -1,82 +1,238 @@
-@Library('pipelib@master') _
-
-def THREADFIX_ID = env.THREADFIX_ID ? env.THREADFIX_ID : '115'
+#!/usr/bin/groovy
 
 node {
   def root = pwd()
-
-  stage('Setup') {
-    git([
-      url: env.GIT_URL ? env.GIT_URL : 'https://github.com/venicegeo/pz-logger',
-      branch: "master"
-    ])
-  }
-
-  stage('Archive') {
-    withGolang {
-      sh """
-        # install metalinter
-        go get -u github.com/alecthomas/gometalinter
-        gometalinter --install
-
-        # build ourself, and go there
-        go get github.com/venicegeo/pz-logger
-        cd \$GOPATH/src/github.com/venicegeo/pz-logger
-
-        go test -v -coverprofile=${root}/logger.cov -coverpkg github.com/venicegeo/pz-logger/logger github.com/venicegeo/pz-logger/logger
-        go tool cover -func=${root}/logger.cov -o ${root}/logger.cov.txt
-
-        cd ${root}
-
-        cp \$GOPATH/bin/pz-logger ${root}/pz-logger
-        tar -cvzf pz-logger.tgz \
-            pz-logger \
-            *.cov \
-            *.cov.txt \
-            glide.lock \
-            glide.yaml
-      """
+  def mvn = tool 'M3'
+  def golangTool = tool 'golang_1.7'
+  def zapHome = tool 'ZAProxy_v2_5_0'
+  def nodejs = tool 'NodeJS_6'
+  def appvers = sh(script: "git describe --long --tags --always | sed 's/\\./-/'g", returnStdout: true)
+  appvers = appvers.trim()
+  def appName = "pz-logger-${appvers}"
+  
+    // Getting the GO environment ready along with defining GIT credentials
+    stage('Setup') {
+        deleteDir()
+        // github is open gitlab requires credentials to clone
+        if(env.GITLAB_CREDS) {
+          git url: "${env.GIT_URL}", branch: "${env.GIT_BRANCH}", credentialsId: "${env.GITLAB_CREDS}"
+        } else {
+          git url: "${env.GIT_URL}", branch: "${env.GIT_BRANCH}"
+        }
+        
+        withEnv([
+           "PATH+=${golangTool}/bin:${root}/gopath/bin",
+            "GOROOT=${golangTool}",
+            "GOPATH=${root}/gopath"
+        ]) {
+            sh """
+              mkdir -p ${root}/gopath/bin ${root}/gopath/pkg ${root}/gopath/src
+              go version
+            """
+        }
     }
-    mavenPush()
-  }
-
-  stage('CI Deploy') {
-    cfPush()
-    zap {
-      threadfixId = THREADFIX_ID
+    
+    if(params.SECENV) {
```

Comment:
You can remove this entire stage if you don't have any special configs you need to add for UC.